packages/release-tool/src/index.ts

```diff
@@ -398,13 +398,53 @@ function formatChangelog(previousReleasedVersion: string, prs: ExtendedGithubPrD
   ].join("\n");
 }
 
+async function getCommitsOfPr(pr: ExtendedGithubPrData) {
+  const commits: components["schemas"]["commit"][] = [];
+
+  for (let page = 1; ; page += 1) {
+    const pageResult = await octokit.request("GET /repos/{owner}/{repo}/pulls/{pull_number}/commits", {
+      owner: "lensapp",
+      repo: "lens",
+      pull_number: pr.number,
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28'
+      },
+      per_page: 100,
+      page,
+    });
+
+    commits.push(...pageResult.data);
+
+    if (pageResult.data.length < 100) {
+      break;
+    }
+  }
+
+  return commits;
+}
+
+async function computeIfPrWasSquashed(pr: ExtendedGithubPrData, commits: components["schemas"]["commit"][]): Promise<boolean> {
+  const { stdout: mergeCommitSubject } = await execFile("git", ["log", "-1", "--format=%s", pr.mergeCommitSha]);
+
+  return commits.every(commit => commit.commit.message !== mergeCommitSubject);
+}
+
 async function cherryPickCommits(prs: ExtendedGithubPrData[]): Promise<void> {
   const rl = createInterface(process.stdin);
   const cherryPickCommit = cherryPickCommitWith(rl);
 
   for (const pr of prs) {
     if (pr.shouldAttemptCherryPick) {
-      await cherryPickCommit(pr.mergeCommitSha);
+      const commits = await getCommitsOfPr(pr);
+      const wasSquashed = await computeIfPrWasSquashed(pr, commits);
+
+      if (wasSquashed) {
+        await cherryPickCommit(pr.mergeCommitSha);
+      } else {
+        for (const commit of commits) {
+          await cherryPickCommit(commit.sha);
```

Comment:
should the order be reversed?

Reply:
I don't think so. Take the following data retrieved about https://github.com/lensapp/lens/pull/7783.

```json
{
  "commits": [
    {
      "authoredDate": "2023-05-23T14:12:18Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "- Add better support for thousands of namespaces\n\n- Add support for mouse only multi-selection\n\nSigned-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "feat: Improve UX of namespace select filter",
      "oid": "e55a32b46be3581d5f405e1332a55f1f70d9a98f"
    },
    {
      "authoredDate": "2023-05-24T16:09:46Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Start moving namespace-select-filter to own folder",
      "oid": "5f293e225c8122a7d2d3d6003c3e79d5e1a5c0e8"
    },
    {
      "authoredDate": "2023-05-24T16:10:27Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Refactor isSelectionKey",
      "oid": "0a62558a4a3a4e23ac804e2a9d27aaf5816548aa"
    },
    {
      "authoredDate": "2023-05-24T16:13:19Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Consolidate namespace-select-filter model into single file",
      "oid": "b83bc189649ab091e504f316bd886bcb8aef3a1e"
    },
    {
      "authoredDate": "2023-05-24T16:13:51Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Rename namespace-select-filter component file",
      "oid": "5f693366d87e98041df57e2d07bf0d379d9867ca"
    },
    {
      "authoredDate": "2023-05-24T18:07:49Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Fixup previous namespace-select-filter unit tests",
      "oid": "1b96c69832f72eca3bbacf889295fb8eaef8ca5d"
    },
    {
      "authoredDate": "2023-05-24T19:14:58Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Add more test for namespace-select-filter to cover new behaviour",
      "oid": "68eb71311111726fe90e20959edeb69ec68ec69b"
    },
    {
      "authoredDate": "2023-05-25T15:25:45Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-25T19:15:32Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Add tests for large number of namespaces",
      "oid": "49445d58f64c5e2860aa32c5a9e245a285f500c3"
    },
    {
      "authoredDate": "2023-05-26T12:41:13Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-26T12:41:13Z",
      "messageBody": "Signed-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Update snapshots due to new behaviour of namespace-select-filter",
      "oid": "961e146b6d7e21f28ee07506d8483eb7124dbc89"
    },
    {
      "authoredDate": "2023-05-26T12:48:33Z",
      "authors": [
        {
          "email": "sebastian@malton.name",
          "id": "MDQ6VXNlcjgyMjUzMzI=",
          "login": "Nokel81",
          "name": "Sebastian Malton"
        }
      ],
      "committedDate": "2023-05-26T14:44:22Z",
      "messageBody": "…filter behaviour\n\nSigned-off-by: Sebastian Malton <sebastian@malton.name>",
      "messageHeadline": "chore: Update smoke test to fix selector due to new namespace-select-…",
      "oid": "14bd0ef18905c8ef999124e38c9f631a2580dac2"
    }
  ]
}
```

The commits are retrieved in order that they appear on the PR, and that is the order that they will be applied to the target branch, so I think that should also be the order that they are applied here.